dispatches/models/fossil_case/properties/thermaloil_properties.py

```diff
@@ -0,0 +1,430 @@
+                     were not provied at the unit model level, the
+                     control volume passes the inlet values as initial
+                     guess.The keys for the state_args dictionary are:
+
+                     flow_mol_comp : value at which to initialize component
+                                     flows (default=None)
+                     pressure : value at which to initialize pressure
+                                (default=None)
+                     temperature : value at which to initialize temperature
+                                  (default=None)
+            outlvl : sets output level of initialization routine
+            state_vars_fixed: Flag to denote if state vars have already been
+                              fixed.
+                              - True - states have already been fixed and
+                                       initialization does not need to worry
+                                       about fixing and unfixing variables.
+                             - False - states have not been fixed. The state
+                                       block will deal with fixing/unfixing.
+            optarg : solver options dictionary object (default=None)
+            solver : str indicating whcih solver to use during
+                     initialization (default = 'ipopt')
+            hold_state : flag indicating whether the initialization routine
+                         should unfix any state variables fixed during
+                         initialization (default=False).
+                         - True - states varaibles are not unfixed, and
+                                 a dict of returned containing flags for
+                                 which states were fixed during
+                                 initialization.
+                        - False - state variables are unfixed after
+                                 initialization by calling the
+                                 relase_state method
+
+        Returns:
+            If hold_states is True, returns a dict containing flags for
+            which states were fixed during initialization.
+        '''
+        # init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+        # solve_log = idaeslog.getSolveLogger(self.name, outlvl,
+        #                                     tag="properties")
+
+        if state_vars_fixed is False:
+            # Fix state variables if not already fixed
+            flags = fix_state_vars(blk, state_args)
+            for k in blk.keys():
+                if degrees_of_freedom(blk[k]) != 0:
+                    raise Exception("State vars fixed but degrees of freedom "
+                                    "for state block is not zero during "
+                                    "initialization.")
+
+        else:
+            # Check when the state vars are fixed already result in dof 0
```

Comment:
Not crucial. I think the for loop can be placed outside if condition. Correct? If yes, pulling it out might simplify the code a bit.
```python
if state_vars_fixed is False:
    flags = ...

for k in blk.keys():
    if ...
```

Reply:
fixed this, thanks